Sources/SwiftDocC/Model/Rendering/Variants/JSONPointer.swift

```diff
@@ -87,36 +87,78 @@ public struct JSONPointer: Codable, CustomStringConvertible, Equatable {
-    }
-    
-    /// Unescaped a path component of a JSON pointer.
-    static func unescape(_ pointerPathComponents: String) -> String {
-        applyEscaping(pointerPathComponents, shouldUnescape: true)
+    private static func escaped(_ pathComponents: [String]) -> String {
+        // This code is called quite frequently for mixed language content.
+        // Optimizing it has a measurable impact on the total documentation build time.
+        
+        var string: [UTF8.CodeUnit] = []
+        string.reserveCapacity(
+            pathComponents.reduce(0) { acc, component in
+                acc + 1 /* the "/" separator */ + component.utf8.count
+            }
+        )
+        
+        for component in pathComponents {
+            // The leading slash and component separator
+            string.append(forwardSlash)
+            
+            // The escaped component
+            for char in component.utf8 {
+                switch char {
+                case tilde:
+                    string.append(contentsOf: escapedTilde)
+                case forwardSlash:
+                    string.append(contentsOf: escapedForwardSlash)
+                default:
+                    string.append(char)
+                }
+            }
+        }
+        
+        return String(decoding: string, as: UTF8.self)
     }
     
-    /// Applies an escaping operation to the path component of a JSON pointer.
-    /// - Parameters:
-    ///   - pointerPathComponent: The path component to escape.
-    ///   - shouldUnescape: Whether this function should unescape or escape the path component.
-    /// - Returns: The escaped value if `shouldUnescape` is false, otherwise the escaped value.
-    private static func applyEscaping(_ pointerPathComponent: String, shouldUnescape: Bool) -> String {
-        EscapedCharacters.allCases
-            .reduce(pointerPathComponent) { partialResult, characterThatNeedsEscaping in
-                partialResult
-                    .replacingOccurrences(
-                        of: characterThatNeedsEscaping[
-                            keyPath: shouldUnescape ? \EscapedCharacters.escaped : \EscapedCharacters.rawValue
-                        ],
-                        with: characterThatNeedsEscaping[
-                            keyPath: shouldUnescape ? \EscapedCharacters.rawValue : \EscapedCharacters.escaped
-                        ]
-                    )
+    private static func unescaped(_ escapedRawString: String) -> [String] {
+        escapedRawString.removingLeadingSlash.components(separatedBy: "/").map {
+            // This code is called quite frequently for mixed language content.
+            // Optimizing it has a measurable impact on the total documentation build time.
+            
+            var string: [UTF8.CodeUnit] = []
+            string.reserveCapacity($0.utf8.count)
+            
+            var remaining = $0.utf8[...]
+            while let char = remaining.popFirst() {
```

Comment:
Because I need to iterate over pairs of characters to identify "~0" and "~1".

At first I though I could iterate over the character's pairwise but when I tried it I realized that it meant that `"A~0B"` would iterate over `(A, ~)`, `(~, 0)`, `(0, B)` which that made the loop more complicated because I needed to keep state from the `(~, 0)` iteration so that the `(0, B)` iteration wouldn't append the `0`.